### PR TITLE
Corrected the order of DM Logic and Ban Action in ban.ts

### LINKED ISSUE #606: [BUG] - Ban command sends DM too late
# Bug Report

## Describe the bug

The `ban` command attempts to send a DM to the user *after* banning them from the server. This causes an error because the user no longer shares a server with Becca.

## Expected behavior

The `ban` command logic needs to be switched such that the DM is sent before the user is banned.

### PR DESCRIPTION
**Pull request by Zipher007**
## Description:

In this, the code for the DM Logic and the Ban action is switched and put in the right order for correct execution.

## Related Issue:

Closes #606

### CODE REVIEW

File: src/commands/moderation/ban.ts [hunk 1]
```diff
@@ -160,13 +160,14 @@ const ban: CommandInt = {
...
         await Becca.sendMessageToLogsChannel(guild, banEmbed);
 
-        // Ban the user.
-        await memberToBanMentioned.ban({ reason });
-
         // Send a message to the user.
         await userToBanMentioned.send(
           `**[Ban]** ${author.toString()} has banned you for the following reason: ${reason}`
         );
+
```
Comment (nhcarrigan) [inline src/commands/moderation/ban.ts:167]:
```suggestion

```

File: src/commands/moderation/ban.ts [hunk 2]
```diff
@@ -160,13 +160,14 @@ const ban: CommandInt = {
...
-        await memberToBanMentioned.ban({ reason });
-
         // Send a message to the user.
         await userToBanMentioned.send(
           `**[Ban]** ${author.toString()} has banned you for the following reason: ${reason}`
         );
+        
+        // Ban the user.
+        await memberToBanMentioned.ban({ reason });
+
```
Comment (nhcarrigan) [inline src/commands/moderation/ban.ts:170]:
```suggestion

```

**Review by nhcarrigan — CHANGES_REQUESTED** [review]
Go ahead and commit these suggestions to fix the linter errors.

**Comment by Zipher007** [comment]
are these linter errors due the spaces in the lines ?

**Comment by nhcarrigan** [comment]
They are, yes - the empty lines should be empty so the linter wanted the spaces removed.

**Comment by Zipher007** [comment]
> They are, yes - the empty lines should be empty so the linter wanted the spaces removed.

Oh I see. In the tests also, it was telling to delete the spaces!
Now I understand the importance of seeing the tests for a particular project !!

**Comment by Zipher007** [comment]
Thank you so much @nhcarrigan  for guiding me!  
Feels really good after helping, even if it was a small issue😃

**Comment by nhcarrigan** [comment]
Glad to help! I'll pull this down sometime today to test it. :)

**Comment by Zipher007** [comment]
> Glad to help! I'll pull this down sometime today to test it. :)

Hope it works fine now !!

**Review by nhcarrigan — APPROVED** [review]

**Pull request merged**